### DISCUSSION TITLE
Mojo source code

### DISCUSSION THREAD

**Discussion by s3drik** [modular/modular#373] [Q&A] [answered]
Mojo isn't open source but can we know in which programming language written mojo

**Comment by mojodojodev** [comment, marked as answer]
@Saddibek1 the parser is currently written in C++, and they would like to self host eventually: https://mojodojo.dev/mojo_team_answers.html#self-hosting. It also lowers to [MLIR](https://mlir.llvm.org/) so could say it's built on top of that.

**Reply by s3drik** [reply]
Python is written in C
Superset of python is written in superset of C?)
